packages/graph/hash_graph/lib/graph/src/store/mod.rs

```diff
@@ -379,6 +379,21 @@ pub trait EntityStore: for<'q> crud::Read<PersistedEntity, Query<'q> = Expressio
         entity_id: Option<EntityId>,
     ) -> Result<PersistedEntityIdentifier, InsertionError>;
 
+    /// Creates new entities.
+    ///
+    /// # Errors:
+    ///
+    /// - if the [`EntityType`] doesn't exist
+    /// - if the entities are not valid with respect to the specified [`EntityType`]
+    /// - if the account referred to by `owned_by_id` does not exist
+    /// - if an [`EntityId`] was supplied and already exists in the store
+    async fn create_entities(
```

Comment:
The main worry I have is that this is being reviewed under the context of its utility for tests, and we'll see these methods and expose them publicly. (We already know we have a need for batch inserts). These methods aren't being made with all the considerations we're going to need to make them under.

For example, we know:
- We need to figure out a way to expose transactions on the public API
- We need to figure out a way to insert entities and links at the same time for referential integrity
- We need to figure out a way to batch insert many things at once (possibly including types)
- We're going to be implementing a lot more validation (for example actually checking the entity blob against its entity type)

I'm especially concerned about the lack of links here (which are an unfortunate omission in the current benchmarking setup so it's understandable).

Reply:
Putting these functions inside of the bench would be possible. I added them to the library to avoid exposing more database internals than needed. Happy to change it, though.

Reply:
I'd be more comfortable with them inside bench.

Reply:
Another problem is that of entities having multiple entity types. I definitely don't want this in the main code for now

Reply:
I tried moving it to `bench` but without exposing much internals (like `VersionId`, which we really don't want to expose) it's not possible.

Reply:
Can you put it behind a non-documented feature-flag (perhaps it should be named something related to being internal or private, **_rather than_** "batched_inserts") and then enable it in the bench Cargo.toml?

Reply:
Added a feature flag for the new functionality, have hidden the functions and put a warning on the flag.